Kotlin/demo/src/main/res/layout/activity_main.xml

```diff
@@ -1,43 +1,119 @@
 <?xml version="1.0" encoding="utf-8"?>
-<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
     android:layout_width="match_parent"
     android:layout_height="match_parent"
-    android:orientation="vertical"
-    android:padding="24dp">
+    android:padding="24dp"
+    tools:ignore="rtlHardcoded">
 
     <EditText
         android:id="@+id/etInput"
-        android:layout_width="match_parent"
+        android:layout_width="0dp"
         android:layout_height="wrap_content"
-        android:layout_gravity="center_horizontal"
+        android:autofillHints="@null"
         android:gravity="center_horizontal"
         android:hint="@string/hint_blurhash"
         android:inputType="text"
-        android:singleLine="true"
+        android:maxLines="1"
         android:text="LEHV6nWB2yk8pyo0adR*.7kCMdnj"
-        android:textColor="@color/colorAccent" />
+        android:textColor="@color/textColor"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent"
+        tools:ignore="HardcodedText" />
 
-    <TextView
-        android:id="@+id/tvDecode"
-        android:layout_width="wrap_content"
+    <Button
+        android:id="@+id/btnDecode"
+        style="@style/Widget.AppCompat.Button.Colored"
+        android:layout_width="0dp"
         android:layout_height="wrap_content"
-        android:layout_gravity="center_horizontal"
         android:layout_marginTop="12dp"
-        android:background="@color/colorPrimary"
-        android:elevation="8dp"
-        android:paddingStart="12dp"
-        android:paddingTop="8dp"
-        android:paddingEnd="12dp"
-        android:paddingBottom="8dp"
         android:text="@string/title_button_decode"
-        android:textColor="@color/colorAccent"
-        android:textSize="16sp" />
+        android:textColor="@color/textColor"
+        android:textSize="16sp"
+        app:layout_constraintEnd_toStartOf="@id/btnEncode1"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@id/etInput" />
+
+    <Button
+        android:id="@+id/btnEncode1"
+        style="@style/Widget.AppCompat.Button.Colored"
+        android:layout_width="0dp"
+        android:layout_height="wrap_content"
+        android:layout_marginLeft="8dp"
```

Comment:
`ConstraintLayout` pretty much uses `start` and `end`. You can change the margin here to reflect that (i.e. to `marginStart` and `marginEnd`. With this we won't be needing `tools:ignore="rtlHardcoded"` as well, iirc.

Reply:
@hangduykhiem `marginStart` minimum API is 17, but this demo app is API 14.
I've added `marginRight`, and removed `tools:ignore="rtlHardcoded"`